src/foam/nanos/auth/User.js

```diff
@@ -229,36 +228,27 @@ foam.CLASS({
       section: 'administrative'
     },
     {
-      class: 'FObjectProperty',
-      of: 'foam.nanos.auth.Phone',
-      name: 'phone',
+      class: 'foam.core.PhoneNumberPropertyValidationRefinement',
+      name: 'phoneNumber',
       documentation: 'Personal phone number.',
-      factory: function() {
-        return this.Phone.create();
-      },
-      view: { class: 'foam.u2.detail.VerticalDetailView' },
       section: 'personal'
     },
     {
-      class: 'String',
-      name: 'phoneNumber',
-      transient: true,
-      documentation: `Omits properties of the phone number object and returns
-        the phone number.`,
-      expression: function(phone) {
-        return phone.number;
-      },
+      class: 'Boolean',
+      name: 'phoneNumberVerified',
+      writePermissionRequired: true,
       section: 'personal'
     },
     {
-      class: 'FObjectProperty',
-      of: 'foam.nanos.auth.Phone',
+      class: 'foam.core.PhoneNumberPropertyValidationRefinement',
```

Comment:
```suggestion
      class: 'PhoneNumber',
```

Reply:
Do you mean that our refinement (PhoneNumberPropertyValidationRefinement) would overwrite the original definition of PhoneNumber? This means PhoneNumber for example can only be refined once?! The semantics of refinement in foam should be clarified, or maybe it is but I don't know where.

Reply:
No, but I see the misunderstanding now. Refinements in FOAM modify/update the model being refined, they don't replace it. So in this way you can define a model and refine it later to add features/enhancements to it.

That's what we're doing in this PR. We're refining the PhoneNumber model (which represents a property whose value is a phone number, not a phone number itself) to add validation information to it. Other parts of the code base do refinements to add other features, such as Element.js setting default views for properties.